pkg/credentials/service_account_credentials.go

```diff
@@ -62,6 +63,58 @@ func NewCredentialBulder(client client.Client, config *v1.ConfigMap) *Credential
 	}
 }
 
+func (c *CredentialBuilder) CreateStorageSpecSecretVolumeAndEnv(namespace string, storageKey string,
+	storageSecretName string, overrideParams map[string]string, container *v1.Container, volumes *[]v1.Volume) error {
+	secret := &v1.Secret{}
+	err := c.client.Get(context.TODO(), types.NamespacedName{Name: storageSecretName,
+		Namespace: namespace}, secret)
+	if err != nil {
+		log.Error(err, "Failed to find secret", "SecretName", storageSecretName)
+		return err
+	}
+	if storageData, ok := secret.Data[storageKey]; ok {
+		var storageDataJson map[string]string
+		err := json.Unmarshal([]byte(storageData), &storageDataJson)
+		if err != nil {
+			return err
+		}
+		// override secret values if parameters are provided
+		for key, element := range overrideParams {
+			storageDataJson[key] = element
+		}
+		if storageDataJson["type"] == "s3" {
+			if endpointURL, ok := storageDataJson["endpoint_url"]; ok {
+				container.Env = append(container.Env, v1.EnvVar{
+					Name:  "AWS_ENDPOINT_URL",
+					Value: endpointURL,
+				})
+			}
+			if accessKeyID, ok := storageDataJson["access_key_id"]; ok {
+				container.Env = append(container.Env, v1.EnvVar{
+					Name:  "AWS_ACCESS_KEY_ID",
+					Value: accessKeyID,
+				})
+			}
+			if secretAccessKey, ok := storageDataJson["secret_access_key"]; ok {
+				container.Env = append(container.Env, v1.EnvVar{
+					Name:  "AWS_SECRET_ACCESS_KEY",
+					Value: secretAccessKey,
+				})
+			}
+			if secretRegion, ok := storageDataJson["region"]; ok {
+				container.Env = append(container.Env, v1.EnvVar{
+					Name:  "AWS_DEFAULT_REGION",
+					Value: secretRegion,
+				})
+			}
+			if secretBucket, ok := storageDataJson["bucket"]; ok {
+				container.Args[0] = strings.Replace(container.Args[0], "<bucket-placeholder>", secretBucket, -1)
```

Comment:
this bucket placeholder thing is kind of nasty :) but I guess it is somewhat temporary